functions/Remove-DbaDbTableData.ps1

```diff
@@ -0,0 +1,326 @@
+
+    .EXAMPLE
+        PS C:\> Get-DbaDatabase -SqlInstance localhost -Database TestDb1, TestDb2  | Remove-DbaDbTableData -Table dbo.Test -BatchSize 1000000 -LogBackupPath E:\LogBackups -Confirm:$false
+
+        Removes data from the dbo.Test table in the TestDb1 and TestDb2 databases on the local SQL instance. The deletes are dones in batches of 1000000 rows each and the log backups are written to E:\LogBackups.
+
+    .EXAMPLE
+        PS C:\> $server, $server2 | Remove-DbaDbTableData -Database TestDb -Table dbo.Test -BatchSize 1000000 -LogBackupPath E:\LogBackups -Confirm:$false
+
+        Removes data from the dbo.Test table in the TestDb database on the SQL instances represented by $server and $server2. The deletes are dones in batches of 1000000 rows each and the log backups are written to E:\LogBackups.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [ValidateRange(1, 1000000000)][int]$BatchSize = 100000,
+        [string]$Table,
+        [string]$DeleteSql,
+        [string]$LogBackupPath,
+        [string]$LogBackupTimeStampFormat,
+        [string[]]$AzureBaseUrl,
+        [string]$AzureCredential,
+        [parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
+        [switch]$EnableException
+    )
+
+    begin {
+        if ((Test-Bound Table) -and (Test-Bound DeleteSql)) {
+            Stop-Function -Message "You must specify either -Table or -DeleteSql, but not both. See the command description for more details."
+            return
+        }
+
+        if (-not $Table -and -not $DeleteSql) {
+            Stop-Function -Message "You must specify either -Table or -DeleteSql. See the command description for more details."
+            return
+        }
+
+        if ((Test-Bound BatchSize) -and (Test-Bound DeleteSql)) {
+            Stop-Function -Message "When using -DeleteSql the -BatchSize param cannot be used. See the command description for more details."
+            return
+        }
+
+        if (Test-Bound DeleteSql) {
+            if ($DeleteSql -inotmatch "top") {
+                Stop-Function -Message "To use the -DeleteSql param you must specify the TOP (N) clause in the DELETE statement. See the command description for more details."
+                return
+            }
+
+            if ($DeleteSql -inotmatch "delete") {
+                Stop-Function -Message "The -DeleteSql param must be a DELETE statement with a TOP (N) clause. See the command description for more details."
+                return
+            }
+        }
+
+        if (Test-Bound LogBackupPath) {
+            $null = Test-ExportDirectory -Path $LogBackupPath
```

Comment:
Here Test-DbaPath should be used instead. Test-ExportDirectory tests if the path is there from the perspective of the account running PS, Test-DbaPath instead does it via the SQL service account. If the result is False, a warning should be raised and the function stopped.

Reply:
ok, thanks, I'll make that change.